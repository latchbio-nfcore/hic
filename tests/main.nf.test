nextflow_pipeline {

    name "Test pipeline"
    script "../main.nf"
    tag "PIPELINE"
    config "conf/test.config"

    test("Run with profile test") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            assert workflow.success

            // TODO
            // assertAll(
            // )
        }
    }

    test("Should split fastqs") {

        when {
            params {
                outdir = "$outputDir"
                split_fastq = true
            }
        }

        then {
            assert workflow.success

            // TODO
            // assertAll(
            // )
        }
    }
}
