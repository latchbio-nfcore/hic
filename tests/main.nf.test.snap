{
    "split_software_versions": {
        "content": [
            "{BOWTIE2_ALIGN={bowtie2=2.5.2, samtools=1.18, pigz=2.6}, BOWTIE2_ALIGN_TRIMMED={bowtie2=2.5.2, samtools=1.18, pigz=2.6}, BOWTIE2_BUILD={bowtie2=2.5.2}, COMBINE_MATES={python=3.9.12}, COOLER_BALANCE={cooler=0.9.2}, COOLER_CLOAD={cooler=0.9.2}, COOLER_DUMP={cooler=0.9.2}, COOLER_MAKEBINS={cooler=0.9.2}, COOLER_ZOOMIFY={cooler=0.9.2}, COOLTOOLS_EIGSCIS={cooltools=0.5.1}, COOLTOOLS_INSULATION={cooltools=0.5.1}, CUSTOM_GETCHROMSIZES={getchromsizes=1.16.1}, FASTQC={fastqc=0.12.1}, GET_RESTRICTION_FRAGMENTS={python=3.9.12}, GET_VALID_INTERACTION={python=3.9.12}, HICPRO2PAIRS={pairix=0.3.7}, HIC_FIND_TADS={hicexplorer=3.7.2}, HIC_PLOT_DIST_VS_COUNTS={hicexplorer=3.7.2}, MERGE_BOWTIE2={samtools=1.15.1}, MERGE_STATS={python=3.9.12}, MERGE_VALID_INTERACTION={sort=8.3}, SPLIT_COOLER_DUMP={cooler=null}, TRIM_READS={gzip=1.1}, Workflow={nf-core/hic=v2.2.0dev}}"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-26T19:31:20.226377845"
    },
    "software_versions": {
        "content": [
            "{BOWTIE2_ALIGN={bowtie2=2.5.2, samtools=1.18, pigz=2.6}, BOWTIE2_ALIGN_TRIMMED={bowtie2=2.5.2, samtools=1.18, pigz=2.6}, BOWTIE2_BUILD={bowtie2=2.5.2}, COMBINE_MATES={python=3.9.12}, COOLER_BALANCE={cooler=0.9.2}, COOLER_CLOAD={cooler=0.9.2}, COOLER_DUMP={cooler=0.9.2}, COOLER_MAKEBINS={cooler=0.9.2}, COOLER_ZOOMIFY={cooler=0.9.2}, COOLTOOLS_EIGSCIS={cooltools=0.5.1}, COOLTOOLS_INSULATION={cooltools=0.5.1}, CUSTOM_GETCHROMSIZES={getchromsizes=1.16.1}, FASTQC={fastqc=0.12.1}, GET_RESTRICTION_FRAGMENTS={python=3.9.12}, GET_VALID_INTERACTION={python=3.9.12}, HICPRO2PAIRS={pairix=0.3.7}, HIC_FIND_TADS={hicexplorer=3.7.2}, HIC_PLOT_DIST_VS_COUNTS={hicexplorer=3.7.2}, MERGE_BOWTIE2={samtools=1.15.1}, MERGE_STATS={python=3.9.12}, MERGE_VALID_INTERACTION={sort=8.3}, SPLIT_COOLER_DUMP={cooler=null}, TRIM_READS={gzip=1.1}, Workflow={nf-core/hic=v2.2.0dev}}"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-26T19:28:03.999972948"
    },
    "Run with profile test": {
        "content": [
            37,
            [
                "SRR4292758.2000_compartments.cis.bw:md5,d7b3015f4856ef058a870056f9c3bdf1",
                "SRR4292758.2000_compartments.cis.lam.txt:md5,e05a7cc3c0af218a5cf6f3ad6ffa8da5",
                "SRR4292758.2000_compartments.cis.vecs.tsv:md5,0d1d4c5fd45d2400a781069bb93d2b76"
            ],
            [
                "SRR4292758_distcount.png:md5,5314c9ad16a1e857682fde4e584103bc",
                "SRR4292758_distcount.txt:md5,bac196d56bcb489c2ef6fe2ea26e07c9"
            ],
            [
                [
                    "SRR4292758_0_bwt2pairs.bam:md5,dc752ae9feb57f0ff2daa3a215956915"
                ],
                [
                    [
                        "SRR4292758.mRSstat:md5,99af5a2e6136a5e003d667afbf438707",
                        "SRR4292758.mpairstat:md5,281aa85283e57308187c33a5bad7dc82",
                        "SRR4292758.null.mmapstat:md5,33dff67a7684ffe44d5dfaccf60d6aad",
                        "SRR4292758_allValidPairs.mergestat:md5,b0a81a4d69ea8e44495a714ac87701e8"
                    ]
                ],
                [
                    "SRR4292758.allValidPairs:md5,55e00a772766779f5d5c2bf2b6fcef6d",
                    [
                        "SRR4292758_contacts.pairs.gz:md5,201d59212cf443fe5da09dbddeef8677",
                        "SRR4292758_contacts.pairs.gz.px2:md5,6275247d930753121aa026870864aa25"
                    ]
                ]
            ],
            [
                "multiqc.log:md5,90a7ccd82c900d4a5aa08d3d3da772c5",
                "multiqc_citations.txt:md5,76bcbb4e4103209b523ed9da8fef74bc",
                "multiqc_data.json:md5,c71ad3b9813fcd2c301eb95bc850e568",
                "multiqc_fastqc.txt:md5,64b89edc94607362b7591cb821b28b93",
                "multiqc_general_stats.txt:md5,892258a0f91b4d21d95e2c5e1ff1cc08",
                "multiqc_hicpro.txt:md5,67513d15d5a00c08d93fd6a7fbd189fa",
                "multiqc_software_versions.txt:md5,8700e2b122c2107aff2a88dbd18933ec",
                "multiqc_sources.txt:md5,de6736507ee9f445e58287ed10090aba"
            ],
            [
                [
                    "SRR4292758.1000_balanced_hicfindtads_boundaries.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "SRR4292758.1000_balanced_hicfindtads_boundaries.gff:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "SRR4292758.1000_balanced_hicfindtads_domains.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "SRR4292758.1000_balanced_hicfindtads_score.bedgraph:md5,da38c91d9b23820e290d2c987f27ab70",
                    "SRR4292758.1000_balanced_hicfindtads_tad_score.bm:md5,81b33e5cd3ff14efac94c2065ac68fe1",
                    "SRR4292758.1000_balanced_hicfindtads_zscore_matrix.cool:md5,3203ee7258462d0c2891e7b62c0bca04"
                ],
                [
                    "SRR4292758.1000_balanced_insulation.tsv:md5,fd0a37e94600e9da4c8de0c58f07e3ed"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-26T19:28:04.028638511"
    },
    "Should split fastqs": {
        "content": [
            37,
            [
                "SRR4292758.2000_compartments.cis.bw:md5,d7b3015f4856ef058a870056f9c3bdf1",
                "SRR4292758.2000_compartments.cis.lam.txt:md5,e05a7cc3c0af218a5cf6f3ad6ffa8da5",
                "SRR4292758.2000_compartments.cis.vecs.tsv:md5,0d1d4c5fd45d2400a781069bb93d2b76"
            ],
            [
                "SRR4292758_distcount.png:md5,5314c9ad16a1e857682fde4e584103bc",
                "SRR4292758_distcount.txt:md5,bac196d56bcb489c2ef6fe2ea26e07c9"
            ],
            [
                [
                    "SRR4292758_0_bwt2pairs.bam:md5,676047426d024788971139d92d2ab097"
                ],
                [
                    [
                        "SRR4292758.mRSstat:md5,17bfb0c8be6c0940c2863d147c1ea824",
                        "SRR4292758.mpairstat:md5,281aa85283e57308187c33a5bad7dc82",
                        "SRR4292758.null.mmapstat:md5,5a72e08a7b8098eeb6e1331c8b866b15",
                        "SRR4292758_allValidPairs.mergestat:md5,b0a81a4d69ea8e44495a714ac87701e8"
                    ]
                ],
                [
                    "SRR4292758.allValidPairs:md5,55e00a772766779f5d5c2bf2b6fcef6d",
                    [
                        "SRR4292758_contacts.pairs.gz:md5,201d59212cf443fe5da09dbddeef8677",
                        "SRR4292758_contacts.pairs.gz.px2:md5,6275247d930753121aa026870864aa25"
                    ]
                ]
            ],
            [
                "multiqc.log:md5,5e26b89dad43c71657280738996dbd0d",
                "multiqc_citations.txt:md5,76bcbb4e4103209b523ed9da8fef74bc",
                "multiqc_data.json:md5,41cf9afa708b9a7cdaf827f28f38b096",
                "multiqc_fastqc.txt:md5,64b89edc94607362b7591cb821b28b93",
                "multiqc_general_stats.txt:md5,892258a0f91b4d21d95e2c5e1ff1cc08",
                "multiqc_hicpro.txt:md5,b97b7b08c5075b76c99a902bd268095c",
                "multiqc_software_versions.txt:md5,8700e2b122c2107aff2a88dbd18933ec",
                "multiqc_sources.txt:md5,d79093d1f6997a31509a14695dd635ea"
            ],
            [
                [
                    "SRR4292758.1000_balanced_hicfindtads_boundaries.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "SRR4292758.1000_balanced_hicfindtads_boundaries.gff:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "SRR4292758.1000_balanced_hicfindtads_domains.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "SRR4292758.1000_balanced_hicfindtads_score.bedgraph:md5,da38c91d9b23820e290d2c987f27ab70",
                    "SRR4292758.1000_balanced_hicfindtads_tad_score.bm:md5,81b33e5cd3ff14efac94c2065ac68fe1",
                    "SRR4292758.1000_balanced_hicfindtads_zscore_matrix.cool:md5,3cef5afd1125b90bb73dcce77d987129"
                ],
                [
                    "SRR4292758.1000_balanced_insulation.tsv:md5,fd0a37e94600e9da4c8de0c58f07e3ed"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-26T19:31:20.698967004"
    }
}